portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -1765,6 +1765,43 @@ public void completeUserRegistration(
 		}
 	}
 
+	/**
+	 * Sets the user's status to inactive. This will also deactivate their
+	 * personal site.
+	 *
+	 * @param userId the primary key of the user
+	 * @throws PortalException
+	 */
+	public void deactivateUser(long userId) throws PortalException {
+		deactivateUser(userId, true);
+	}
+
+	/**
+	 * Sets the user's status to inactive. Can also optionally deactivate the
+	 * user's personal site.
+	 *
+	 * @param userId the primary key of the user
+	 * @param deactivateSite whether the user's personal site should be
+	 *                       deactivated
+	 * @throws PortalException
+	 */
+	public void deactivateUser(long userId, boolean deactivateSite)
+		throws PortalException {
+
+		updateStatus(
+			userId, WorkflowConstants.STATUS_INACTIVE, new ServiceContext());
+
+		if (!deactivateSite) {
+			User user = getUser(userId);
+
+			Group group = user.getGroup();
+
+			group.setActive(true);
```

Comment:
@drewbrokke, is it possible `group` could be inactive when the `deactivateUser()` call is made? If so, is it desirable that a call to `deactivateUser()` with `deactivateSite = false` would have the effect of making `group` active when it wasn't previously?

Reply:
That's a good question.  I think the assumption in a most cases is that the user's group is active immediately, and stays that way until you deactivate the user.  It's technically possible to deactivate the user's group manually, but it doesn't look like we do that anywhere in the code.

I am very open to suggestions about how we can fix this with a semantic method signature (I really struggled with that in this case).

Reply:
Based on your description, it doesn't sound like much of a concern, but one remedy to completely avoid the possibility of inadvertently activating a deactivated `group` is to call `getActive()` on `group` prior to calling `updateStatus()` to determine whether `group` was active to begin with. Splitting the method seems tricky in terms of naming, as what comes to mind is `deactivateUser()` and `deactivateUserWithSite()` or `deactivateUserAndSite()`, the latter two being undesirable names.

Reply:
Yeah that was exactly my dilemma.  I also didn't love using a boolean parameter to express code paths, but the alternative is multiple, awkwardly-named methods for each type of scenario, which seems more like a job for a helper class rather than a ServiceImpl.

Reply:
At the _very_ least, I will update the documentation with a more accurate description of what really happens inside the method.

Reply:
This may all be a moot point considering I just discovered that we already have a transactional MVCActionCommand class: https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/portlet/bridges/mvc/BaseTransactionalMVCActionCommand.java.  

@JorgeFerrer @brianchandotcom the more I look at the code I moved into the service layer, the more I wish it was still in the action command.  It is very specific to our use-case and not something I would recommend be published as API, especially with the concerns that @noahsherrillwork raised above.  

My suggestion would be to roll back the changes to the service layer and instead have the appropriate `*MVCActionCommand` classes extend off of `BaseTransactionalMVCActionCommand`.  Thoughts?